robber/expect.py

```diff
@@ -35,12 +36,11 @@ def remove_custom_message(cls):
     def register(cls, name, klass):
         cls.matchers[name] = klass
         method = lambda self, other=None, *args: \
-                klass(self.object, other, *args) \
-                    .fail_with(self.message) \
-                    .match()
+                                            klass(self.obj, other, self.is_negated, *args) \
+                                            .fail_with(self.message) \
+                                            .match()
```

Comment:
I think it's better to declare a method here, rather than making a big lambda like this...